apps/04TowersOfHanoi.js

```diff
@@ -10,74 +10,209 @@ var stacks = {
     c: []
 };
 
+var numberOfTurns = 0;
+
 function printStacks() {
     console.log("a: " + stacks.a);
     console.log("b: " + stacks.b);
     console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-    // Your code here
+function printNumberOfTurns(numberOfTurns) {
+    console.log("# of Turns: " + numberOfTurns);
+}
+
+function movePiece(startStack, endStack) {
+
+    var lengthA = stacks.a.length;
+    var lengthB = stacks.b.length;
+    var lengthC = stacks['c'].length;
+
+    //adjust the array to move the last number from the start stack and add to the front of the end stack
+    var lastValueA = stacks.a[lengthA - 1];
+    var lastValueB = stacks.b[lengthB - 1];
+    var lastValueC = stacks.c[lengthC - 1];
+
+    if (startStack === 'a') {
```

Comment:
This looks clean, but a little repetitive. What if we had 5 or 7 or 100 stacks? Would we write 100 `else if` blocks?